__main__.py

```diff
@@ -3,81 +3,149 @@
 import datetime as dt
 import json
 import sys
+from enum import Enum
 from pathlib import Path
-from typing import NoReturn
+from typing import TYPE_CHECKING, Any, ClassVar, NewType, NoReturn, TypedDict, cast
 
 import requests
 from dahlia import dahlia, dprint
+from requests.exceptions import HTTPError
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+
+__version__ = "0.1.0"
+
+
+class Status(str, Enum):
+    ACCEPTED = "Accepted"
+    ACTIVE = "Active"
+    DEFERRED = "Deferred"
+    DRAFT = "Draft"
+    FINAL = "Final"
+    PROVISIONAL = "Provisional"
+    REJECTED = "Rejected"
+    REPLACED = "Replaced"
+    WITHDRAWN = "Withdrawn"
+    SUPERSEDED = "Superseded"
+
 
 COLOR_CODES = {
-    "Accepted": "&2",
-    "Active": "&a",
-    "Deferred": "&6",
-    "Draft": "&8",
-    "Final": "&1",
-    "Provisional": "&5",
-    "Rejected": "&4",
-    "Replaced": "&e",
-    "Withdrawn": "&d",
+    Status.ACCEPTED: "&2",
+    Status.ACTIVE: "&a",
+    Status.DEFERRED: "&6",
+    Status.DRAFT: "&8",
+    Status.FINAL: "&1",
+    Status.PROVISIONAL: "&5",
+    Status.REJECTED: "&4",
+    Status.REPLACED: "&e",
+    Status.WITHDRAWN: "&d",
+    Status.SUPERSEDED: "",  # FIXME Pick color
 }
-LATEST = Path("pep-tracker/latest.json")
+
+LATEST = Path(__file__).parent.resolve() / "latest.json"
 URL = "https://peps.python.org/api/peps.json"
+PEP = NewType("PEP", str)
+
+
+def now() -> dt.datetime:
+    return dt.datetime.now(tz=dt.timezone.utc)
+
+
+class StateDict(TypedDict):
+    version: str
+    data: dict[str, str]
+    time: str
```

Comment:
For type-hinting the `dict` representation of `State`. I'd personally rather use something like attrs + cattrs to do the conversion instead, though.

Reply:
why `data`  is not `dict[PEP, Status]`? since `PEP` is essentially a `str`, it's hashable, so I can't think of any issue with doing that way

Reply:
I think I've tried that, and didn't have a lot of success, but I could try it again.

`Status` won't be automatically recreated, though, and it isn't exactly an `str`. So when I want to go from JSON back to a dict, it'll be read as `str`, and I'll need to manually re-instantiate them. So a `dict[PEP, str]` might work, but a `dict[PEP, Status]` likely won't.